dlt/helpers/weaviate_helper.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Dict, Union
+from dlt.extract.source import DltResource
+
+
+def weaviate_adapter(
+    resource: DltResource,
+    vectorize: Union[List[str], str] = None,
+    tokenization: Dict[str, str] = None,
+) -> DltResource:
+    if vectorize or tokenization:
+        column_hints = {}
+        if vectorize:
+            if isinstance(vectorize, str):
+                vectorize = [vectorize]
+            if not isinstance(vectorize, list):
+                raise ValueError(
+                    "vectorize must be a list of column names or a single "
+                    "column name as a string."
+                )
+
+            for prop in vectorize:
+                column_hints[prop] = {
+                    "name": prop,
+                    "x-vectorize": True,
+                }
+        if tokenization:
+            for prop, method in tokenization.items():
+                if prop in column_hints:
+                    column_hints[prop]["x-tokenization"] = method
+                else:
```

Comment:
there are only a limited number of [possible values](https://weaviate.io/developers/weaviate/config-refs/schema#property-tokenization) for method: 

* word
* lowercase
* whitespace
* field

would it be possible for dlt to enforce this check in case user made a typo?

Reply:
yes, we can add it to our schema definition

Reply:
One alternative is to leave the validation to Weaviate itself. That way there's not need to sync validation code should the list of supported tokenization methods change. Weaviate's error message is very well written:

```
<class 'weaviate.exceptions.UnexpectedStatusCodeException'>
Create class! Unexpected status code: 422, with response body: 
{'code': 606, 'message': 'properties.0.tokenization in body should be one of [word lowercase whitespace field]'}.
```

Reply:
@burnash that's what I was worried about. Would an experiend dlt user know how to interpret which column properties.0 is referring to? Otherwise, if we follow this suggestion, then we should document how the mapping occurs so dlt user can figure it out themselves.

Reply:
@hsm207 good point, let's handle the validation here